### PR TITLE
Log when an eviction occurs

### PR DESCRIPTION
**Pull request by JakeSiFive**
This change causes the shared cache .stderr file to log evictions. It would be nicer to log to .stdout but that is being used as a communication channel between the job-cache daemon and the eviction child process. We should be able to switch the eviction child process over to a thread now which would resolve this issue.

### CODE REVIEW

File: src/job_cache/eviction_policy.cpp [hunk 1]
```diff
@@ -213,9 +222,10 @@ struct LRUEvictionPolicyImpl {
     // Now we remove the threads in the background.
     // TODO: Figure out how many cores we actully have and use a multiple of that
     // NOTE: We don't wait for this to finish
-    std::async(std::launch::async, [dir = cache_dir, jobs_to_remove = std::move(jobs_to_remove)]() {
-      remove_backing_files(dir, jobs_to_remove, 8, 128);
-    });
+    // std::async(std::launch::async, [dir = cache_dir, jobs_to_remove =
+    // std::move(jobs_to_remove)]() {
```
Comment (V-FEXrt) [inline src/job_cache/eviction_policy.cpp:226]: Can you explain / fix the commented code here?

File: src/job_cache/eviction_policy.cpp [hunk 2]
```diff
@@ -213,9 +222,10 @@ struct LRUEvictionPolicyImpl {
...
+    remove_backing_files(cache_dir, jobs_to_remove, 8, 128);
```
Comment (V-FEXrt) [inline src/job_cache/eviction_policy.cpp:227]: Where did `cache_dir` come from? I don't see a new var introduced

File: src/job_cache/job_cache_impl_common.cpp [hunk 3]
```diff
@@ -251,11 +255,17 @@ void remove_backing_files(const std::string &dir, int64_t job_id) {
...
     }
-    unlink_no_fail(entry->name.c_str());
+    files_to_remove.push_back(wcl::join_paths(job_dir, entry->name));
+  }
+  for (const auto &file : files_to_remove) {
+    unlink_no_fail(file.c_str());
   }
+
+  rmdir_no_fail(job_dir.c_str());
 }
```
Comment (V-FEXrt) [inline src/job_cache/job_cache_impl_common.cpp:265]: do you need to rebase?

**Review by V-FEXrt — CHANGES_REQUESTED** [review]

File: src/job_cache/eviction_policy.cpp (hunk 1 above)
Comment (JakeSiFive) [reply src/job_cache/eviction_policy.cpp:226]: Oh right. This is the "async's destructor blocks" issue. I should probably put a bit of work into using an std::thread like I did before instead.

File: src/job_cache/eviction_policy.cpp (hunk 2 above)
Comment (JakeSiFive) [reply src/job_cache/eviction_policy.cpp:227]: previously we had a generalized capture of `dir = cache_dir`

File: src/job_cache/job_cache_impl_common.cpp (hunk 3 above)
Comment (JakeSiFive) [reply src/job_cache/job_cache_impl_common.cpp:265]: Nope! I found bugs in the removal while writing this change. This fixes them.

**Review by V-FEXrt — APPROVED** [review]

**Pull request merged**